src/main/java/com/google/research/bleth/servlets/NewExperimentSimulationServlet.java

```diff
@@ -0,0 +1,124 @@
+import com.google.appengine.api.datastore.TransactionOptions;
+import com.google.research.bleth.simulator.AbstractSimulation;
+import com.google.research.bleth.simulator.AwakenessStrategyFactory;
+import com.google.research.bleth.simulator.MovementStrategyFactory;
+import com.google.research.bleth.simulator.Schema;
+import com.google.research.bleth.simulator.StrategiesMapper;
+import com.google.research.bleth.simulator.TracingSimulation;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A servlet used for creating and running a new simulation associated to an experiment,
+ * and updating experimentsToSimulations database entity (which maps experiments to their associated simulations).
+ */
+@WebServlet("/new-experiment-simulation")
+public class NewExperimentSimulationServlet extends HttpServlet {
+    private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static final Logger log = Logger.getLogger(NewExperimentSimulationServlet.class.getName());
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        log.info("Experiment Simulation Worker Servlet invoked.");
+        log.info("Enqueue Experiment Servlet received an http POST request with params: "
+        + request.getParameterMap().toString());
+
+        // Get request parameters.
+        String experimentId = request.getParameter("experimentId");
+        String simulationDescription = request.getParameter("description");
+        int roundsNum = Integer.parseInt(request.getParameter("roundsNum"));
+        int rowsNum = Integer.parseInt(request.getParameter("rowsNum"));
+        int colsNum = Integer.parseInt(request.getParameter("colsNum"));
+        int beaconsNum = Integer.parseInt(request.getParameter("beaconsNum"));
+        int observersNum = Integer.parseInt(request.getParameter("observersNum"));
+        String beaconMovementStrategyAsString = request.getParameter("beaconMovementStrategy");
+        String observerMovementStrategyAsString = request.getParameter("observerMovementStrategy");
+        String observerAwakenessStrategyAsString = request.getParameter("observerAwakenessStrategy");
+        int awakenessCycle = Integer.parseInt(request.getParameter("awakenessCycle"));
+        int awakenessDuration = Integer.parseInt(request.getParameter("awakenessDuration"));
+        double transmissionThresholdRadius = Double.parseDouble(request.getParameter("transmissionThresholdRadius"));
+
+        // Mapping strategies string to types.
+        StrategiesMapper strategiesMapper = StrategiesMapper.getInstance();
+        MovementStrategyFactory.Type beaconMovementStrategy = strategiesMapper.getMovementStrategy(beaconMovementStrategyAsString);
```

Comment:
Can this fail? How will your system behave if it does?
(Same applies to the next 3 lines)

Reply:
If at this point one of the strategies strings is not valid, a `null` value will be returned from `getMovementStrategy()` / `getAwakenessStrategy()`. Then the Builder's `build()` method will throw a `NullPointerException` with a message indicating the missing argument, e.g "No beacon movement strategy has been set.".

The strategies string are also being validated on client side (in the case where the simulation is not associated with an experiment) or on the enqueuing servlet (in our case - it only uses the default RANDOM strategy, which is valid).

The only change I can think of is modifying the `getMovementStrategy()` / `getAwakenessStrategy()` method to throw an exception rather than returning `null` in such case. If you believe this is necessary I will do so (I believe this belongs to a separate PR thought).

@nolaviz